libs/gltfio/include/gltfio/AssetLoader.h

```diff
@@ -0,0 +1,145 @@
+ * clients to load a small set of precompiled ubershader materials to avoid the footprint of the
+ * filamat library on resource-constrained platforms.
+ *
+ * Example usage:
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * auto engine = Engine::create();
+ * auto loader = AssetLoader::create(engine);
+ *
+ * // Parse the glTF content and create Filament entities.
+ * std::vector<uint8_t> content(...);
+ * FilamentAsset* asset = loader->createAssetFromJson(content.data(), content.size());
+ * content.clear();
+ *
+ * // Load buffers and textures from disk.
+ * ResourceLoader({engine, ".", true}).loadResources(asset);
+ *
+ * // Obtain the simple animation interface.
+ * Animator* animator = asset->getAnimator();
+ *
+ * // Free the glTF hierarchy as it is no longer needed.
+ * asset->releaseSourceData();
+ *
+ * // Add renderables to the scene.
+ * scene->addEntities(asset->getEntities(), asset->getEntityCount());
+ *
+ * // Execute the render loop and play the first animation.
+ * do {
+ *      animator->applyAnimation(0, time);
+ *      animator->updateBoneMatrices();
+ *      if (renderer->beginFrame(swapChain)) {
+ *          renderer->render(view);
+ *          renderer->endFrame();
+ *      }
+ * } while (!quit);
+ *
+ * loader->destroyAsset(asset);
+ * loader->destroyMaterials();
+ * AssetLoader::destroy(&loader);
+ * Engine::destroy(&engine);
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ */
+class AssetLoader {
+public:
+
+    /**
+     * Creates an asset loader for the given engine.
+     *
+     * The engine is held weakly, used only for the creation and destruction of Filament objects.
+     * The optional name component manager can be used to assign names to renderables.
+     */
+    static AssetLoader* create(filament::Engine* engine, utils::NameComponentManager* = nullptr);
+
+    /**
+     * Frees the loader.
+     *
+     * This does not not automatically free the cache of materials (see destroyMaterials), nor
+     * does it free the entities for created assets (see destroyAsset).
+     */
+    static void destroy(AssetLoader** loader);
+
+    /**
+     * Takes a pointer to the contents of a JSON-based glTF 2.0 file and returns a bundle
+     * of Filament objects. Returns null on failure.
+     */
+    FilamentAsset* createAssetFromJson(uint8_t const* bytes, uint32_t nbytes);
+
+    /**
+     * Takes a pointer to the contents of a GLB glTF 2.0 file and returns a bundle
+     * of Filament objects. Returns null on failure.
+     */
+    FilamentAsset* createAssetFromBinary(uint8_t const* bytes, uint32_t nbytes);
+
+    /** Destroys the given asset and all of its associated Filament objects. */
+    void destroyAsset(const FilamentAsset* asset);
+
+    /**
+     * Enables or disables shadows on all subsequently loaded assets.
+     *
+     * Initially, loaded assets will cast and receive shadows.
+     */
+    void castShadowsByDefault(bool enable);
```

Comment:
We have setters on `Renderable` for those two properties. I think we should remove them from this API and just users iterate over the loaded renderables if they want to toggle shadow casting/receiving instead.

Reply:
Sounds good.

Reply:
> The ability to not rely on libfilamat
Yes this is already a TODO in the big AssetLoader comment.

Reply:
Ok, removed these methods